Domains/Jobs/Exceptions/JobException.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file Domains/Jobs/Exceptions/JobException.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class JobException
+ * @ingroup domains
+ *
+ * @brief Exception for Job domain
+ */
+
+namespace PKP\Domains\Jobs\Exceptions;
+
+use Exception;
+
+class JobException extends Exception
+{
+    public const INVALID_PAYLOAD = 'invalid.payload';
```

Comment:
This is just to avoid coding the same string when creating the exception, right? I don't mind if the same string appears in a few places, but either way, this would need to be a little more helpful to someone who ran into it.

Reply:
Indeed. Also, I'm using this kind of logic for the past 5 years, it's easier to find where is called (`JobException::INVALID_PAYLOAD`), instead the pure string (invalid.payload). Became a unique placeholder.

Reply:
OK, but if end users encounter this message they won't have any idea what's happening (unless they search the code). And a more helpful message can be searched the same way as this unique placeholder. Could you make this a human-readable string? If we adopt this kind of placeholder, we should consider it as a general approach, rather than just adding it here. Otherwise the codebase can get really inconsistent with lots of new ideas that are never adopted beyond experiments.

Reply:
Changed the output message.